packages/atomic-react/src/components/stencil-generated/react-component-lib/createComponent.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+import {
+  attachProps,
+  createForwardRef,
+  dashToPascalCase,
+  isCoveredByReact,
+  mergeRefs,
+} from './utils';
+
+export interface HTMLStencilElement extends HTMLElement {
+  componentOnReady(): Promise<this>;
+}
+
+interface StencilReactInternalProps<ElementType> extends React.HTMLAttributes<ElementType> {
+  forwardedRef: React.RefObject<ElementType>;
+  ref?: React.Ref<any>;
+}
+
+export const createReactComponent = <
```

Comment:
there's a lot of potential here. if we can't wrap Stencil components, maybe calling its lifecycle events 1:1 from React's may work.

Reply:
Indeed, we can use the stencil code as inspiration, but may want to modify things. I did not find the vanilla stencil-react output a good experience, but I encourage you to try the branch out yourselves,